.github/workflows/android.yml

```diff
@@ -0,0 +1,469 @@
+name: Android CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout to Branch
+      uses: actions/checkout@v3
+    - name: Echo build job
+      run: echo "Build job done"
+    - name: set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: gradle
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build with Gradle
+      run: ./gradlew build
+    - name: Build with Gradle
+      run: ./gradlew assembleDebug
+    - name: Upload APK
+      uses: actions/upload-artifact@v2
+      with:
+        name: APK
+        path: app/build/outputs/apk/debug/**.apk
+  
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout to Branch
+      uses: actions/checkout@v3
+    - name: set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: gradle
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Run Linter and Test
+      run: ./gradlew check
+    - name: Run lint Ktlint
+      run: ./gradlew ktlintCheck --continue
+    - name: Upload Reports
+      uses: actions/upload-artifact@v2
+      with:
+        name: Test-Reports
+        path: app/build/reports
+    - name: Run Tests
+      run: ./gradlew test
+    - name: Run Test Coverage
+      run: ./gradlew testDebugUnitTestCoverageVerification
+    - name: Generate JaCoCo Badge
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+        generate-branches-badge: true
+        jacoco-csv-file: "**/testDebugUnitTestCoverage.csv"
+    - name: upload coverage report
+      uses: actions/upload-artifact@v3.1.2
+      with:
+        name: coverage report
+        path: app/build/reports/jacoco/testDebugUnitTestCoverage/html
```

Comment:
You have not add separate job for coverage report since we need to deploy the docker image with next major release to docker hub (as per assignment Task 1 point 3).